tools/reporter/index.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const util = require('util')
+const HackerOneClient = require('./HackerOneClient')
+const NswgReporter = require('./NswgReporter')
+
+const USER = process.env.NSWG_BB_API_USERNAME
+const PASS = process.env.NSWG_BB_API_PASSWORD
```

Comment:
Do not store credentials in environment variables.

Reply:
@ChALkeR:
1. would we want to source a config file from the system this is running on?
a-la ~/.aws/credentials ?
2. if we will run this on CI, which I know you're not too big fan of atm due to the security implications for that, how would we provide the relevant creds?

Reply:
The problem with environment variables is that they are not treated as secure, and on-disk file storage with right permissions is [better](https://twitter.com/joepie91/status/954748152917250049).

Some personal evidence:
* npm passwords and api tokens [ended up being published in the registry packages](https://github.com/ChALkeR/notes/blob/master/Do-not-underestimate-credentials-leaks.md#express-mime-types-component-) _because_ they were leaking to the environment variables and another tool did not treat env vars as secure and saved them in `config.gypi`, which ended up being published. For `express` this happened _more than once_.
* Including the tokens on a CI, in any form, also has risks — I [was able](https://github.com/ChALkeR/notes/blob/master/Stealing-Travis-secure-variables.md) to trick Travis CI (and some others) into exposing encrypted credentials.

Re: reading from fs and CI — what would be the usecases for this tool? I assumed manual execution of this tool with explicit confirmation of the action taken for each module. Because even the «automatically open a generic security issue invite on the related module's repo» action without human confirmation sounds highly dubious. I do not think that things should be published _automatically_ from CI.

So, perhaps just ask the credentials each time from the CLI, optionally password manager integration?

About the CI part — could we securely limit it's access to only the public reports somehow? Automatically opening pull requests to this repo based on published reports should be fine.

Reply:
Thanks for the context and elaborate information, insightful as always! I actually did have your notes gist starred already from a previous read :-)

Having an interactive prompt on the CLI sounds good.
I'll make appropriate changes.

About CI and automation around tooling - we're not there yet. I haven't finalized requirements and actually think we'd benefit more from having a webhook originating from H1 platform to us (some service somewhere) but since H1 aren't supporting a custom webhook then this isn't relevant atm. Anyway I think we'll take this gradually and securely to see how the process works and eases human intervention so we can more easily handle the load.